src/Common/FileCache.cpp

```diff
@@ -698,18 +663,12 @@ LRUFileCache::ReserveResult LRUFileCache::tryReserveForQuery(const Key & key, si
 
         for (auto & cell : to_evict)
         {
-            auto file_segment = cell->file_segment;
-            if (file_segment)
-            {
-                query_context->remove(file_segment->key(), file_segment->offset(), cell->size(), cache_lock);
-
-                std::lock_guard<std::mutex> segment_lock(file_segment->mutex);
-                remove(file_segment->key(), file_segment->offset(), cache_lock, segment_lock);
-            }
+            if (auto file_segment = cell->file_segment)
+                remove_file_segment(file_segment, cell->size());
         }
 
         query_context->reserve(key, offset, size, cache_lock);
-        return ReserveResult::FITS_IN_QUERY_LIMIT_NEED_RESERVE_FROM_MAIN_LIST;
+        return true;
     }
```

Comment:
The reason for the exception is that the wrong status was returned here before.

Reply:
yes, this will call reserve twice.